optaplanner-core/src/main/java/org/optaplanner/core/api/domain/variable/VariableListener.java

```diff
@@ -27,10 +30,14 @@
  * It can change its shadow variable(s) on multiple entity instances
  * (for example: an arrivalTime change affects all trailing entities too).
  * <p>
- * Each {@link ScoreDirector} has a different {@link VariableListener} instance, so it can be stateful.
- * If it is stateful, it must implement {@link StatefulVariableListener}.
+ * It is recommended that implementations be kept stateless.
+ * If state must be implemented, implementations may need to override the default methods
+ * ({@link #resetWorkingSolution(ScoreDirector)}, {@link #close()}).
+ *
+ * @param <Solution_> @{@link PlanningSolution} holding the @{@link PlanningEntity} on which the variable is declared
```

Comment:
This "the PlanningEntity" kinda implies there there is only one planning entity class and only one planning entity.

Also, I suggest we standarize the javadoc sentence for this generic parameter across all javadocs. Most use this sentence (that we might want to improve):
`* @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation`